src/main/java/gregtech/api/color/ColoredBlockContainer.java

```diff
@@ -0,0 +1,96 @@
+    public static void registerContainer(@NotNull ColoredBlockContainer container) {
+        CONTAINERS.add(Objects.requireNonNull(container, "A null ColoredBlockContainer cannot be registered!"));
+    }
+
+    public static @NotNull ColoredBlockContainer getContainer(@NotNull World world, @NotNull BlockPos pos,
+                                                              @NotNull EnumFacing facing,
+                                                              @NotNull EntityPlayer player) {
+        for (ColoredBlockContainer container : CONTAINERS) {
+            if (container.isValid(world, pos, facing, player)) {
+                return container;
+            }
+        }
+
+        return NullColorContainer.NULL_CONTAINER;
+    }
+
+    @ApiStatus.Internal
+    public static void registerCEuContainers() {
+        registerContainer(new GTPipeColorContainer());
+        registerContainer(new MTEColorContainer());
+        if (Mods.AppliedEnergistics2.isModLoaded()) {
+            registerContainer(new AE2ColorContainer());
+        }
+        registerContainer(new VanillaColorContainer());
+    }
+
+    public abstract boolean isValid(@NotNull World world, @NotNull BlockPos pos, @NotNull EnumFacing facing,
+                                    @NotNull EntityPlayer player);
+
+    public abstract boolean setColor(@NotNull World world, @NotNull BlockPos pos, @NotNull EnumFacing facing,
+                                     @NotNull EntityPlayer player, @Nullable EnumDyeColor newColor);
+
+    public boolean setColor(@NotNull World world, @NotNull BlockPos pos, @NotNull EnumFacing facing,
+                            @NotNull EntityPlayer player, int newColor) {
+        return false;
+    }
+
+    public abstract boolean removeColor(@NotNull World world, @NotNull BlockPos pos, @NotNull EnumFacing facing,
+                                        @NotNull EntityPlayer player);
+
+    public abstract @Nullable EnumDyeColor getColor(@NotNull World world, @NotNull BlockPos pos,
+                                                    @NotNull EnumFacing facing, @NotNull EntityPlayer player);
+
+    public int getColorInt(@NotNull World world, @NotNull BlockPos pos, @NotNull EnumFacing facing,
+                           @NotNull EntityPlayer player) {
+        EnumDyeColor dyeColor = getColor(world, pos, facing, player);
+        return dyeColor == null ? -1 : dyeColor.colorValue;
+    }
+
+    public boolean colorMatches(@NotNull World world, @NotNull BlockPos pos, @NotNull EnumFacing facing,
+                                @NotNull EntityPlayer player, @Nullable EnumDyeColor color) {
+        return getColor(world, pos, facing, player) == color;
+    }
+
+    public boolean colorMatches(@NotNull World world, @NotNull BlockPos pos, @NotNull EnumFacing facing,
+                                @NotNull EntityPlayer player, int color) {
+        return getColorInt(world, pos, facing, player) == color;
+    }
+
+    public boolean supportsARGB() {
```

Comment:
This is named `supportsARGB` but the MC colors don't provide `A` at all. Should this just be renamed to `supportsRGB`?